addon/utils/store.js

```diff
@@ -21,11 +21,30 @@ export default EmberObject.extend({
   },
 
   initialsFor(properties) {
-    let key = hash(properties);
+    let hashableProp = this._ensureHashable(properties);
+    let key = md5(hashableProp);
     return this.get('cache')[key] || this._create(key, properties);
   },
 
   _create(key, properties) {
     return this.get('cache')[key] = this.get('generator').generate(properties);
+  },
+
+  _ensureHashable(properties) {
+    if (typeOf(properties) !== 'object') {
+      // this should never happen, but just in case
+      return properties.toString();
+    }
+
+    let pKeys = Object.keys(properties).sort();
```

Comment:
Just curious, why we are sorting it?

Reply:
@Exelord Just an attempt at increasing the cache hit rate by standardizing key order.

{a: 'b', c: 'd'} should equal {c: 'd', a: 'b'}.

Reply:
Right! I will also add tests for that case.